eth/tracers/live/supply.go

```diff
@@ -0,0 +1,245 @@
+package live
+
+import (
+	"encoding/json"
+	"log"
+	"math/big"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth/tracers/directory"
+	"github.com/ethereum/go-ethereum/params"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+func init() {
+	directory.LiveDirectory.Register("supply", newSupply)
+}
+
+type SupplyInfo struct {
+	Delta       *big.Int `json:"delta"`
+	Reward      *big.Int `json:"reward"`
+	Withdrawals *big.Int `json:"withdrawals"`
+	Burn        *big.Int `json:"burn"`
+
+	// Block info
+	Number     uint64      `json:"blockNumber"`
+	Hash       common.Hash `json:"hash"`
+	ParentHash common.Hash `json:"parentHash"`
+}
+
+type supplyTxCallstack struct {
+	calls []supplyTxCallstack
+	burn  *big.Int
+}
+
+type Supply struct {
+	delta       SupplyInfo
+	txCallstack []supplyTxCallstack // Callstack for current transaction
+	logger      *log.Logger
+
+	// Check if genesis has been processed
+	hasGenesisProcessed bool
+}
+
+func newSupply(ctx *directory.LiveTracerContext) (core.BlockchainLogger, error) {
+	// Store traces in a rotating file
+	loggerOutput := &lumberjack.Logger{
+		Filename: filepath.Join(ctx.OutputPath, "supply.jsonl"),
+		MaxSize:  200, // megabytes
```

Comment:
What do you think of adding a flag for this as well @s1na?

Reply:
Sure but I think we should change the config and make it similar to native tracers. I.e. we can set the tracer name, aand a JSON string that will be passed to the specific tracer for its configuration. Actually now that we have a concrete use-case I'll go ahead and do that in my branch. Then you can merge and use it.

Reply:
Pushed. So now in the CLI can do `--vmtrace.config '{"path": "bla", "maxSize": "200mb"}'` and parse that in the constructor of the supply tracker.

Reply:
Thanks @s1na. Much better, here is the change https://github.com/s1na/go-ethereum/pull/21/commits/27bc3911b7fb15768d6e32f674bc319f523132e8